### PR TITLE
Consolidate REST API specifications and publish under Apache 2.0 license

### LINKED ISSUE #69392: REST API specifications are consolidated and published under Apache 2.0 license
We want to continue to encourage folks to develop third-party clients and integrations with Elasticsearch under flexible licensing terms. To facilitate this, we want it to be clear that the Elasticsearch API itself is available under a permissive license. We have an existing notion of an API specification i the form of JSON files describing the various endpoints. These are currently licensed under the new defaults of SSPL or Elastic license. We want to do two main things here:

1. Consolidate the REST API specifications in a single place. This means moving the specs that are currently in `x-pack` into the `rest-api-spec` project so we no longer make a distinction between x-pack and non-x-pack.
2. Clearly designate the `rest-api-spec` project as being Apache 2.0 licensed, and update the published artifact accordingly.

### PR DESCRIPTION
**Pull request by mark-vieira**
This pull request introduces a number of refactoring all intended to simply how we deal with our REST API specifications and YAML tests. This includes the following:

1. The REST API specifications (JSON files) for x-pack APIs that resided under `/x-pack/plugin/src/test` have been moved to the `/rest-api-spec` project so they are now colocated with the "core" specifications. This means we no longer distinguish between an API being "oss" or "x-pack".
2. With the change in (1) the `RestResourcesPlugin` has been updated to reflect this and now when including API specs there is a single `include` method there, rather than separate ones for "core" and "xpack". All these references have also been updated in various build scripts.
3. The "core" REST API tests have been moved from `/rest-api-spec/src/main` to `/rest-api-spec/src/yamlRestTest` to more appropriately align with source set conventions. This also means these tests are are no longer included in the `rest-api-spec` artifact published to maven central.
4. The "xpack" REST API tests have been moved from `/x-pack/plugin/src/test` to `x-pack/plugin/src/yamlRestTest` for similar reasons to (3).
5. The `javaRestTest` source set in `:x-pack:plugin` has been removed, and the two tests that used to live there moved to the appropriate owning module.
6. A handful of other unused resources in `:x-pack:plugin` have also been removed.

Closes #69392

### CODE REVIEW

**Review by breskeby — APPROVED** [review]

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
cc @Mpdreamz 

We'll want to wait until we've merged https://github.com/elastic/infra/pull/27377 and the clients team has migrated to using the rest resources artifact before merging this as otherwise we'll break their workflows due to all the moved files.

File: x-pack/plugin/security/qa/security-trial/src/javaRestTest/resources/roles.yml
Comment (jakelandis) [inline x-pack/plugin/security/qa/security-trial/src/javaRestTest/resources/roles.yml:14]: is this change intentional ?

File: x-pack/plugin/security/qa/security-trial/build.gradle
Comment (jakelandis) [inline x-pack/plugin/security/qa/security-trial/build.gradle:25]: Is this change intentional ?

**Review by jakelandis — APPROVED** [review]
Changes look good.  Really nice having it all in one place. Is this planned to be merged back to 7.x ? If not, it could be a pretty big source of merge conflict issues on back ports.

File: x-pack/plugin/security/qa/security-trial/build.gradle
Comment (mark-vieira) [reply x-pack/plugin/security/qa/security-trial/build.gradle:25]: Yes, a test was moved into this project and needed the additional user.

**Comment by mark-vieira** [comment]
Yes, the intention is to backport, which will be a pain but it's gotta be done.

**Pull request merged**
